### PR TITLE
support content-range header

### PR DESCRIPTION
**Pull request by Nightmare-MY**
i don't know some code like matches[1]! whether  call on a null

### CODE REVIEW

**Comment by Nightmare-MY** [comment]
@googlebot I signed it!

**Comment by kevmoo** [comment]
Please add tests!

**Comment by Nightmare-MY** [comment]
> 请添加测试！

I added

**Review by kevmoo — CHANGES_REQUESTED** [review]
This is a great start, but we need to do more.

Need to reply with the 'Accept-Ranges: bytes' for all file requests.

See more here https://developer.mozilla.org/en-US/docs/Web/HTTP/Range_requests

**Comment by Nightmare-MY** [comment]
> This is a great start, but we need to do more.
> 
> Need to reply with the 'Accept-Ranges: bytes' for all file requests.
> 
> See more here https://developer.mozilla.org/en-US/docs/Web/HTTP/Range_requests

i added,is that so?

**Comment by kevmoo** [comment]
I think we need a LOT more testing here. What are we supported to do with invalid ranges, etc.

CC @natebosch

**Comment by Nightmare-MY** [comment]
> I think we need a LOT more testing here. What are we supported to do with invalid ranges, etc.
> 
> CC @natebosch

now i added.

**Comment by Nightmare-MY** [comment]
now i added more test,Is that right?

**Comment by Nightmare-MY** [comment]
hello,Is there anything I haven't done yet? @kevmoo

**Comment by kevmoo** [comment]
Thoughts @natebosch ?

File: lib/src/static_handler.dart [hunk 1]
```diff
@@ -190,12 +190,64 @@ Future<Response> _handleFile(Request request, File file,
...
+          );
+        }
+
+        // Override Content-Length with the actual bytes sent.
+        headers[HttpHeaders.contentLengthHeader] = (end - start + 1).toString();
+
+        // Set 'Partial Content' status code.
+        headers[HttpHeaders.contentRangeHeader] = 'bytes $start-$end/$length';
+        // Pipe the 'range' of the file.
+        if (request.method == 'HEAD') {
```
Comment (natebosch) [inline lib/src/static_handler.dart:236]: https://datatracker.ietf.org/doc/html/rfc7233#section-3.1

> A server MUST ignore a Range header field received with a request method other than GET.

**Review by kevmoo — APPROVED** [review]

**Comment by kevmoo** [comment]
I'm going to land as-is and do a follow-up. Thanks @Nightmare-MY !!

File: lib/src/static_handler.dart (hunk 1 above)
Comment (kevmoo) [reply lib/src/static_handler.dart:236]: I just tried this w/ https://storage.googleapis.com/dart-archive/channels/stable/release/2.13.4/sdk/dartsdk-macos-x64-release.zip.sha256sum

And it IS honored for HEAD responses. Returned with a `content-length` of `5`

Comment (natebosch) [reply lib/src/static_handler.dart:236]: I cannot repro.

```
curl --head https://storage.googleapis.com/dart-archive/channels/stable/release/2.13.4/sdk/dartsdk-macos-x64-release.zip.sha256sum -H "Range: 0-5" | grep content-length

x-goog-stored-content-length: 96
content-length: 96
```

Comment (kevmoo) [reply lib/src/static_handler.dart:236]:
```
curl --head https://storage.googleapis.com/dart-archive/channels/stable/release/2.13.4/sdk/dartsdk-macos-x64-release.zip.sha256sum -H "Range: bytes=0-5" | grep content-length

x-goog-stored-content-length: 96
content-length: 6
```

Comment (kevmoo) [reply lib/src/static_handler.dart:236]: must specify `bytes=0-5`

Comment (natebosch) [reply lib/src/static_handler.dart:236]: 😳 Ah yeah I did miss the `bytes=`...

This is a hard call to make. The spec is unambiguous, but it does look like buggy servers are fairly common and from a little searching it seems some clients depend on those bugs 😞 

@jakemac53 - WDYT?

Comment (kevmoo) [reply lib/src/static_handler.dart:236]: For comparison: https://raw.githubusercontent.com/dart-lang/shelf_static/range_side_by_side/example/files/dart_sdk_test_example.sha256sum

**Comment by kevmoo** [comment]
This is published now as v1.1 @Nightmare-MY

**Comment by Nightmare-MY** [comment]
thanks! @kevmoo

**Pull request merged**